client/components/VerticalBar/VerticalBar.tsx

```diff
@@ -1,9 +1,13 @@
 import { Box } from '@rocket.chat/fuselage';
-import React, { FC, ComponentProps, memo } from 'react';
+import React, { FC, ComponentProps, memo, CSSProperties } from 'react';
 
 import { useLayoutContextualBarPosition, useLayoutSizes } from '../../providers/LayoutProvider';
 
-const VerticalBar: FC<ComponentProps<typeof Box>> = ({ children, ...props }) => {
+const VerticalBar: FC<ComponentProps<typeof Box> & { overridePosition?: CSSProperties['position'] }> = ({
```

Comment:
Why are these changes necessary? I would leave changing this generic component as a last resort. What exactly are you trying to fix here? Maybe we can find a better solution

Reply:
Indeed, since the vertical bar is not made to fit inside a `filterByText` component. It is made to work on a page. The solution here is to move the component up the component tree and use it on the `CurrentChatPage.tsx`. You will probably have to move some code that handles the changes to the params up as well to prevent multiple sources changing the same state, leading to unaccounted side effects.

Reply:
I'd actually done something similar in my previous commit [here](https://github.com/RocketChat/Rocket.Chat/pull/25000/commits/e35b6c4710119998347d0b20e4f77a30668c75df) but I was running into infinite re-render loop since every time I update the custom fields in the parent component (i.e. `CurrentChatsPage`), it was re-rendering the `GenericTable` component, that was internally re-rendering the `FilterByText`, that was finally triggering the re-rendering of the main component (i.e. `CurrentChatsRoute`) and this was leading to an infinite loop.

Reply:
or we also could separate the form from the table (since it is a huge one that we barely support) and let the params handle the table data.